libbeat/common/file/rotator.go

```diff
@@ -271,191 +389,154 @@ func (r *Rotator) dirMode() os.FileMode {
-	}
-	if r.redirectStderr {
-		RedirectStandardError(r.file)
+		return d
 	}
-	return nil
-}
 
-// appendToFile opens an existing log file for appending. Unlike openFile it
-// does not call MkdirAll because it is an error for the file to not already
-// exist.
-func (r *Rotator) appendToFile() error {
-	var err error
-	r.file, err = os.OpenFile(r.filename, os.O_WRONLY|os.O_APPEND, r.permissions)
-	if err != nil {
-		return errors.Wrap(err, "failed to append to existing file")
-	}
-	if r.redirectStderr {
-		RedirectStandardError(r.file)
+	// continue from last file
+	if len(files) != 0 {
+		if len(files) == 1 {
+			d.currentFilename = files[0]
+		} else {
+			d.SortModTimeLogs(files)
+			d.currentFilename = files[len(files)-1]
+		}
 	}
-	return nil
-}
 
-func (r *Rotator) closeFile() error {
-	if r.file == nil {
-		return nil
-	}
-	err := r.file.Close()
-	r.file = nil
-	r.size = 0
-	return err
+	return d
 }
 
-func (r *Rotator) purgeOldBackups() error {
-	if r.intervalRotator != nil {
-		return r.purgeOldIntervalBackups()
-	}
-	return r.purgeOldSizedBackups()
+func (d *dateRotator) ActiveFile() string {
+	return d.currentFilename
 }
 
-func (r *Rotator) purgeOldIntervalBackups() error {
-	files, err := filepath.Glob(r.filename + "*")
-	if err != nil {
-		return errors.Wrap(err, "failed to list existing logs during rotation")
-	}
-
-	if len(files) > int(r.maxBackups) {
-
-		// sort log filenames numerically
-		r.intervalRotator.SortIntervalLogs(files)
-
-		for i := len(files) - int(r.maxBackups) - 1; i >= 0; i-- {
-			f := files[i]
-			_, err := os.Stat(f)
-			switch {
-			case err == nil:
-				if err = os.Remove(f); err != nil {
-					return errors.Wrapf(err, "failed to delete %v during rotation", f)
-				}
-			case os.IsNotExist(err):
-				return errors.Wrapf(err, "failed to delete non-existent %v during rotation", f)
-			default:
-				return errors.Wrapf(err, "failed on %v during rotation", f)
-			}
-		}
+func (d *dateRotator) Rotate(reason rotateReason, rotateTime time.Time) error {
+	if d.log != nil {
+		d.log.Debugw("Rotating file", "filename", d.currentFilename, "reason", reason)
 	}
 
+	d.currentFilename = d.filenamePrefix + "-" + rotateTime.Format(d.format)
 	return nil
 }
 
-func (r *Rotator) purgeOldSizedBackups() error {
-	for i := r.maxBackups; i < MaxBackupsLimit; i++ {
-		name := r.backupName(i + 1)
-
-		_, err := os.Stat(name)
-		switch {
-		case err == nil:
-			if err = os.Remove(name); err != nil {
-				return errors.Wrapf(err, "failed to delete %v during rotation", name)
-			}
-		case os.IsNotExist(err):
-			return nil
-		default:
-			return errors.Wrapf(err, "failed on %v during rotation", name)
+func (d *dateRotator) RotatedFiles() []string {
+	files, err := filepath.Glob(d.filenamePrefix + "*")
+	if err != nil {
+		if d.log != nil {
+			d.log.Debugw("failed to list existing logs: %+v", err)
 		}
 	}
+	if len(files) == 1 && files[0] == d.ActiveFile() {
```

Comment:
if there are multiple files are we removing active file? if not we should not remove it from here as well

Reply:
You are right. It is leftover from a different approach. Thanks for catching it!